processor/src/main/resources/org/mapstruct/ap/internal/model/OptionalMappingMethod.ftl

```diff
@@ -0,0 +1,39 @@
+<#--
+
+    Copyright MapStruct Authors.
+
+    Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+
+-->
+<#-- @ftlvariable name="" type="org.mapstruct.ap.internal.model.OptionalMappingMethod" -->
+<#if overridden>@Override</#if>
+<#lt>${accessibility.keyword} <@includeModel object=returnType/> ${name}(<#list parameters as param><@includeModel object=param/><#if param_has_next>, </#if></#list>)<@throws/> {
+    if ( ${sourceParameter.name} == null ) {
+        <#if resultType.optionalType>
+            <#-- regardless of mapNullToDefault value, never return null for Optional resultType -->
+            return Optional.empty();
+        <#else>
+            <#if !mapNullToDefault>
+                return null;
+            <#else>
+                <#-- TODO: this should return the default value for whatever the returnType is -->
+                return null;
```

Comment:
Not sure how to return the "default" type for when the `resultType` is not `Optional` (e.g. when it is a collection). Preferably mapping `Optional<Collection> == null` => `Empty Collection` (though I do recognize that no one should be passing a `null` to an `Optional`, but we should prob still handle this case regardless.

Also `Optional.<Collection>empty()` should probably also map to `Empty Collection` when `mapNullToDefault` is true, but that logic too is missing below.